src/modules/site-v2/templates/tools/phenotype_database/submit-traits.html

```diff
@@ -130,14 +137,27 @@ <h5 class="modal-title" id="confirmationModalLabel">Thank you</h5>
 
         // If trait is passed as a string value, look up its fields in the phenotype table
         if (trait !== null && typeof trait === 'string') {
-            const el = document.getElementById(`trait-data-${trait}`);
-            trait = {
-                'name':              trait,
-                'dataset':           dataset,
-                'trait_name_caendr': trait,
-                'description_short': el.dataset.description,
-                'species':           el.dataset.species,
-            }
+            queryTraitByName(trait)
+                // Re-call the function using the trait value instead of the name
+                // Response check prevents infinite recursion
+                .then((response) => {
+                    if (response !== null && typeof response === 'string') throw Error(`Invalid response for trait ${id}: "${response}"`);
+                    select_trait(id, response, dataset);
+                })
+                .fail((error) => {
+                    // If an error message was received, display it, otherwise use a fallback message
+                    console.error(error);
+                    const resp = error.responseJSON
+                    if (resp && resp.message && false) {
```

Comment:
can we clarify the intention here? assuming `resp` is object/null, `resp.message` is undefined/string/object then using `false` appears to convolute the intention with mixed types.

Reply:
Hm... I think I threw `false` in here because I wanted to keep this structure (which I use elsewhere for error handling) but I only want to flash the default message?  Usually it's `if (resp && resp.message)`, to just check if the `message` field exists.  Definitely weird though, I'll clean it up.